src/AppComponents/PizzaForm.jsx

```diff
@@ -0,0 +1,261 @@
+import { customAlphabet, nanoid } from 'nanoid';
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import shortid from 'shortid';
+import * as yup from 'yup';
+
+function PizzaForm(props) {
+  const history = useHistory();
+  const [pizzaOptions, setPizzaOptions] = useState({
+    size: ['Small', 'Medium', 'Large', 'Extra Large'],
+    sauce: ['Original Red', 'Garlic Ranch', 'BBQ Sauce', 'Spinach Alfredo'],
+    toppings: [
+      'Pepperoni',
+      'Sausage',
+      'Canadian Bacon',
+      'Spicy Italian Sausage',
+      'Garlic Chicken',
+      'Onions',
+      'Green Pepper',
+      'Diced Tomatos',
+      'Black Olives',
+      'Roasted Garlic',
+      'Artichoke Hearts',
+      'Three Cheese',
+      'Pineapple',
+      'Extra Cheese',
+    ],
+    substitute: ['Gluten Free Crust'],
+  });
+
+  const [order, setOrder] = useState({
+    name: '',
+    size: '',
+    sauce: '',
+    toppings: [],
+    substitute: '',
+    instructions: '',
+  });
+
+  const [isSubmitEnabled, setSubmitEnabled] = useState(false);
+
+  const formSchema = yup.object().shape({
+    name: yup.string().required('Name for order is required'),
+    size: yup.string().required('Size is required'),
+    sauce: yup.string(),
+    toppings: yup.array().max(10, "Can't have more than 10 toppings"),
+    substitute: yup.string(),
+    instructions: yup.string(),
+  });
+
+  useEffect(() => {
+    formSchema.isValid(order).then((valid) => {
+      setSubmitEnabled(valid);
+    });
+  }, [order]);
+
+  const changeInput = (evt) => {
+    if (evt.target.type !== 'radio' || evt.target.type !== 'checkbox') {
+      setOrder({ ...order, [evt.target.name]: evt.target.value });
+    }
+
+    if (evt.target.type === 'radio') {
+      setOrder({ ...order, [evt.target.name]: evt.target.value });
+    }
+
+    if (evt.target.name === 'toppings') {
+      const toppings = [...order.toppings];
+      if (toppings.length < 10) {
+        if (evt.target.checked === true) {
+          toppings.push(evt.target.value);
+          setOrder({ ...order, toppings: toppings });
+        } else {
+          const newToppings = toppings.filter(
+            (topping) => topping !== evt.target.value
+          );
+          setOrder({ ...order, toppings: newToppings });
+        }
+      } else {
+        setOrder({ ...order });
+      }
+    }
+
+    if (evt.target.name === 'substitute') {
+      if (evt.target.checked === true) {
+        setOrder({ ...order, substitute: evt.target.value });
+      } else {
+        setOrder({ ...order, toppings: '' });
+      }
+    }
+  };
+
+  const submitForm = (evt) => {
+    evt.preventDefault();
```

Comment:
There should be an axios call somewhere in here to the fake API that you used in one of the last projects. it's "https://reqres.in/api/users", which I found on this Canvas page https://lambdaschool.instructure.com/courses/296/pages/objective-3-make-post-requests-to-pass-data-collected-from-a-form-to-a-database?module_item_id=503791